components/login/SignUpForm.js

```diff
@@ -1,77 +1,81 @@
+    setSignupPassConf(newValue);
+  };
 
-    const handlePassConf = (newValue) => {
-        setSignupPassConf(newValue);
-    };
-
-    const handleSignup = (e) => {
-        e.preventDefault();
-        if (signupPassword !== signupPassConf) {
-            setSignupNotification(
-                'Password and password confirmation do not match'
-            )
-            setSignupPassword('');
-            setSignupPassConf('');
-            return null;
-        }
-        createUserWithEmailAndPassword(auth, signupEmail, signupPassword)
-            .catch((err) => {
-                console.log(err.code, err.message)
-                setSignupNotification(err.message)
-            });
+  const handleSignUp = (e) => {
+    e.preventDefault();
+    if (signupPassword !== signupPassConf) {
+      setSignupNotification("Password and password confirmation do not match!");
+      setSignupPassword("");
+      setSignupPassConf("");
+      return null;
     }
+    createUserWithEmailAndPassword(auth, signupEmail, signupPassword)
+      .then((userCredential) => {
+        const user = userCredential.user;
+        console.log(`The user '${user.uid}' has signed up.`);
+        router.push({
+          pathname: "/[user]",
+          query: {
+            user: user.uid,
+            name: signupEmail.match(/^([^@]*)@/),
+          },
+        });
+      })
+      .catch((err) => {
+        console.log(err.code, err.message);
+        setSignupNotification(err.message);
+      });
+    setSignupEmail("");
+    setSignupPassword("");
+    setSignupPassConf("");
+  };
 
-    return (
-        <div>
-            <h1>Sign up</h1>
-            <form onSubmit={handleSignup}>
-                <EmailInputWithLabel
-                    value={signupEmail}
-                    onChange={handleEmail}
-                />
-                <br />
-                <PasswordInputWithLabel
-                    value={signupPassword}
-                    onChange={handlePassword}
-                    labelTextContent='Password'
-                />
-                <br />
-                <PasswordInputWithLabel
-                    value={signupPassConf}
-                    onChange={handlePassConf}
-                    labelTextContent='Confirm password'
-                />
-                <br />
-                <SubmitButton buttonTextContent='Sign up' />
-                <br />
-                {signupNotification}
-            </form>
-        </div>
-    );
+  return (
+    <form onSubmit={handleSignUp}>
+      <h2>Sign up</h2>
+      <EmailInputWithLabel value={signupEmail} onSetEmail={handleSetEmail}>
+        &#128231;
+      </EmailInputWithLabel>
+      <PasswordInputWithLabel
+        value={signupPassword}
+        onSetPassowrd={handleSetPassword}>
+        &#128273;
+      </PasswordInputWithLabel>
+      <PasswordInputWithLabel
+        isConfirm
+        value={signupPassConf}
+        onSetPassword={handleSetPassConf}>
+        &#128273;
```

Comment:
I think that labels content is essential for the accessibility machine-reading and replacing words with emojis is not the best practice for such a purpose. In order to have a neat and less space-consuming input fields appearance, I would suggest having visible text placeholders inside the input fields describing their purpose (hiding on focusing) instead of labels while having the labels styled invisible just for machine-reading. In this case, having an invisible emoji would have less sense than plain text as label content.

Reply:
Same reply and hope this demonstrate how much time you demand on other people unjustly and you've kept doing since day 1 of he project.